### ISSUE TITLE
Some parameter combinations are skipped during search

### ISSUE THREAD

**Issue by jing-IG** [microsoft/nni#1631]
**Short summary about the issue/question**:
I'm not able to get results for all the possible parameter combinations. It seems that some combinations are skipped by NNI.

**Brief what process you are following**: 
I ran a simple experiment. From WebUI, I saw that results are missing for some combinations of parameters. See attached screenshot. For example, there is no line linking 'SGD' to 'ReLU', and no line linking 'SGD' to 'Tanh' either.

**How to reproduce it**: 
Here is the config file:

> authorName: default
> experimentName: single_var_func
> trialConcurrency: 4
> maxExecDuration: 5h
> maxTrialNum: 5000
> #choice: local, remote, pai
> trainingServicePlatform: local
> searchSpacePath: search_space_quadratic_func.json
> #choice: true, false
> useAnnotation: false
> tuner:
>   #choice: TPE, Random, Anneal, Evolution,
>   #SMAC (SMAC should be installed through nnictl)
>   builtinTunerName: TPE
>   classArgs:
>     #choice: maximize, minimize
>     optimize_mode: maximize
> trial:
>   command: python3 one_var_func_nni.py quadratic
>   codeDir: .
>   gpuNum: 0

And the search space file:

> {
>     "hidden_size_1": {"_type": "choice", "_value": [16]},
>     "hidden_size_2": {"_type": "choice", "_value": [16]},
>     "hidden_size_3": {"_type": "choice", "_value": [8]},
>     "act_func": {"_type": "choice", "_value": ["ReLU", "LeakyReLU", "Sigmoid", "Tanh", "Softplus"]},
>     "learning_rate": {"_type": "uniform", "_value": [0.05, 0.005]},
>     "optimizer": {"_type": "choice", "_value": ["SGD", "Adam", "RMSprop"]}
> }

I expect to see all combinations of `act_func` (5 choices) and `optimizer` (3 choices) to be present in the WebUI results. But I didn't get them. Any insights as to why this happens?

**nni Environment**:
- nni version: `nnictl --version` reports 1.0
- nni mode(local|pai|remote): local
- OS: Ubuntu 18.04
- python version: 3.6.8
- is conda or virtualenv used?: No
- is running in docker?: No

**need to update document(yes/no)**:

**Anything else we need to know**:
![1_IRSVJAj32rIUgWjzlvujvw](https://user-images.githubusercontent.com/17995229/67119312-bf1bb600-f19b-11e9-9250-d20633b8bd1d.jpeg)

**Comment by ultmaster** [comment]
Hi. For your particular case, if you intended to see all the combinations, you should set `builtinTunerName` to `GridSearch`. `TPE` does not guarantee it will see all the combinations (in a limited budget of course). For more details, please refer to https://nni.readthedocs.io/en/latest/Tuner/BuiltinTuner.html

**Comment by jing-IG** [comment]
Wonderful. That works. Thanks a lot!
